client/my-sites/themes/theme-showcase.jsx

```diff
@@ -410,39 +461,57 @@ class ThemeShowcase extends Component {
 							select={ this.onTierSelect }
 						/>
 					) }
-					{ isLoggedIn && (
-						<SectionNav className="themes__section-nav" selectedText={ this.state.tabFilter.text }>
-							<NavTabs>
-								{ Object.values( this.tabFilters )
-									.sort( ( a, b ) => a.order - b.order )
-									.map(
-										( tabFilter ) =>
-											this.shouldShowTab( tabFilter.key ) && (
-												<NavItem
-													key={ tabFilter.key }
-													onClick={ () => this.onFilterClick( tabFilter ) }
-													selected={ tabFilter.key === this.state.tabFilter.key }
-													count={ this.notificationCount( tabFilter.key ) }
-												>
-													{ tabFilter.text }
-												</NavItem>
-											)
+					{ isLoggedIn &&
+						( isNewSearchAndFilter ? (
+							<div className="theme__filters">
+								<ThemesToolbarGroup
+									items={ Object.values( this.tabFilters ).filter( ( tabFilter ) =>
+										this.shouldShowTab( tabFilter.key )
```

Comment:
I guess we're able to filter the `tabFilters` at the beginning and related props changed. Then, we can calculate `selectedKeyIndex` when `onFilterClick` is triggered and get rid of `useEffect` inside `ThemesToolbarGroup`

Reply:
Sorry, I don't quite understand the suggestion 😅 Are you suggesting:
1. Instead of using `shouldShowTab` in JSX, we can pre-filter tabs in `getTabFilters()`?
2. Instead of passing `selectedKey`, we save `tabFilterIndex` in `<ThemeShowcase />` state and pass the index to `<ThemesToolbarGroup />`? If that's the case, we would need to either sync both `tabFilter` and `tabFilterIndex` in the component state, or remove `tabFilter` in favor of `tabFilterIndex`. I prefer to keep using `tabFilter` to avoid refactoring for now. What do you think?

Would appreciate further clarification, thank you!

Reply:
Sorry for being unclear!

> 1 . Instead of using shouldShowTab in JSX, we can pre-filter tabs in getTabFilters()?

Yes! I think the `isJetpackSite`, `isAtomicSite`, and `siteCanInstallThemes` props won't change when the component re-renders, so we might be able to pre-filter tabs instead of filtering on every render.

> 2. Instead of passing selectedKey, we save tabFilterIndex in <ThemeShowcase /> state and pass the index to <ThemesToolbarGroup />? If that's the case, we would need to either sync both tabFilter and tabFilterIndex in the component state, or remove tabFilter in favor of tabFilterIndex. I prefer to keep using tabFilter to avoid refactoring for now. What do you think?

Yes, what I want is to get rid of `useEffect` to calculate `activeIndex`. But now I think `useMemo` is enough 🙂
See https://beta.reactjs.org/learn/you-might-not-need-an-effect#caching-expensive-calculations